### DISCUSSION TITLE
How do I skip some dependencies on a particular triplet

### DISCUSSION THREAD

**Discussion by gracicot** [microsoft/vcpkg#25586] [Q&A] [answered]
I have the following manifest file:
```
{
    "name": "my-project",
    "version-string": "0.1.0",
    "dependencies": [
        "nlohmann-json",
        "glm",
        "fmt",
        "glfw3",
        "vulkan-headers",
        "spirv-tools",
        "spirv-headers",
        "glslang",
        "abseil"
    ]
}
```

The reality is that I only need `glfw3`, `vulkan-headers`, `spirv-tools`, `spirv-headers`, `glslang`, `abseil` when building for the native triplet.

The thing is that I also build my project with emscripten, which I use the emscripten headers instead of the libraries listed above. In that case, I only need `nlohmann-json`, `glm` and `fmt`. Building all the other libraries actually breaks the build since some libraries are not compatible with emscripten and don't actually make sense to be built to wasm.

Is there a way I can skip those libraries when building for the emscripten triplet?

**Comment by Osyotr** [comment]
You can group dependencies into features and use only subset of these features for a particular triplet.
https://github.com/microsoft/vcpkg/blob/master/docs/users/buildsystems/cmake-integration.md#vcpkg_manifest_features

**Comment by dg0yt** [comment, marked as answer]
You can restrict dependencies using a `platform` expression.
https://vcpkg.io/en/docs/users/manifests.html#platform-field
